openml/datasets/dataset.py

```diff
@@ -629,6 +629,8 @@ def _encode_if_category(column):
         elif array_format == "dataframe":
             if scipy.sparse.issparse(data):
                 return pd.DataFrame.sparse.from_spmatrix(data, columns=attribute_names)
+            elif isinstance(data, np.ndarray):
+                return pd.Series(data)
```

Comment:
Hey, what's the reason not to apply `columns=attribute_names` here?

Reply:
Also, how can this become a numpy array in the first place? If I understand this correctly, this is called by `get_data`, which splits off `y` by doing `y = data.iloc[:, targets]` from which point on it should be and stay a series, right?